### DISCUSSION TITLE
[Azerothcore] [DBC] Creating Custom Items DBC Part

### DISCUSSION THREAD

**Discussion by KaizerAce** [azerothcore/azerothcore-wotlk#4238] [Guides & Tips]
Hello everyone,

This will be my first guide ever. I've seen a few people asking about how to create custom items and that its been hard to find updated tutorials... So here we are.

I do not take any credit for the editors I have provided since I did not create them.

**Step 1) Getting required files:**

Download this file, there will be 2 folders inside the winrar. Extract those files.
- DBC Editor 'Allows you to edit DBC files'
- MPQ Editor 'Allows you to create/edit MPQ files'
https://www.mediafire.com/file/1ftk8w55834vxnk/DBC_Editing.rar/file

**Step 2) Creating custom items:**

Create all your custom items in your [item_template](https://www.azerothcore.org/wiki/item_template) table.

EX: I will just duplicate [Thunderfury, Blessed Blade of the Windseeker] weapon and change the entry/name for this tutorial, you can obviously change the items quality/damage etc...
https://imgur.com/4p6uBns

**Step 3) Adding custom items to item.dbc:**

- Open up your DBC Editor 'DBC Editor/WDBX Editor.exe'
- Top left corner press File -> Open File(s) -> Item.dbc 'located in your servers DBC folder'
- Click Open
- Select WOTLK 3.3.5
- In the tope Menu bard find 'Tools -> WotLK Item Import'
- Fill in your DB information
  - Server: 127.0.0.1
  - User: acore 'This is azerothcore's default username, may differ from what you have created'
  - Pass: acore 'This is azerothcore's default password, may differ from what you have created'
  and hit 'Connect'
- Database: acore_world 'Azerothcore default DB name may differ from what you have created'
- Table: item_template
- Template: Trinity
- Press 'Load'
https://imgur.com/fWpG3ao

**Step 4) Making sure custom items have been added:**

Now check that your custom items have been added to the item.dbc
Press the little wheel on where it says 'ID'
Sort Largest to smallest
https://imgur.com/EuIXuEr
Hopefully all your items have been added!
https://imgur.com/9b0euNW
You can now close the DBC editor.

**Step 5) Creating the Patch:**

- Open up your MPQ Editor 'MPQ editor\x64\MPQEditor.exe'
- Top left corner hit NEW MPQ https://imgur.com/RmrCqAK
- Enter the name of the patch in 'Enter the name of the archive to be created'. I will name it 'Patch-Z' https://imgur.com/Zlp30vQ
- Hit Next
- Create an Empty MPQ archive
- Next
- Don't change anything there https://imgur.com/3af0zeK
- Next
- Click Finish https://imgur.com/HIXoPGY
- In the left column right click on your patch 'Patch-Z.mpq' in my case.
- Click New Folder https://imgur.com/KxlM4JI
- Name it DBFilesClient
- Click Ok
- Now Press the big green plus at the top menu bar 'Add Files' https://imgur.com/cFLFLI6
- Located your item.dbc that we have edited in step 4 'Server\Data\dbc\item.dbc' is where mine is located
- Press Open once you have found it
- Now Close the MPQ Editor and located your patch
'MPQ editor\x64\Patch-Z.MPQ

**Step 6) Putting the patch into your wow folder:**

- Located your World Of Warcraft Client's Data folder. https://imgur.com/bWEreNo
'D:\World of Warcraft 3.3.5a (no install)\Data' my location
- Paste your newly created patch inside. https://imgur.com/eFHjfVW
- Delete your cache and make sure to restart your client.

Enjoy your Custom items! EVERY PLAYER will be REQUIRED to have this PATCH inside their CLIENT. Without it the items wont work properly.

Please feel free to ask questions if you are having any issues, and let me know if there is anything that I could expand on or improve this guide!

**Comment by Pixelycia** [comment]
Great guide, thanks, helped me a lot! For future adventurers I will leave my comment here as well.

1. You can skip entire step 5 and inside WDBX Editor click Export > "To MPQ" and just place patch according to step 6.

2. Unfortunately, as for 2024 step 3 does not work anymore (probably because of newer MySQL versions and updated SSL), so you have to add items manually or make a script that will create CSV file and import as CSV.
